src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -89,8 +91,23 @@ static MsQuicApi()
 
         if (OperatingSystem.IsWindows())
         {
-            // Windows ships msquic in the assembly directory.
-            loaded = NativeLibrary.TryLoad(Interop.Libraries.MsQuic, typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory, out msQuicHandle);
```

Comment:
Just for my education, why wasn't `AssemblyDirectory` working? Perhaps it just needed additional flag:

```diff
-            loaded = NativeLibrary.TryLoad(Interop.Libraries.MsQuic, typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory, out msQuicHandle);
+            loaded = NativeLibrary.TryLoad(Interop.Libraries.MsQuic, typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory | DllImportSearchPath.ApplicationDirectory, out msQuicHandle);
```
for single-file?

Reply:
The reason was to workaround #107040.

Reply:
Isn't https://github.com/dotnet/runtime/issues/107040 about making it look at _fewer_ locations and this is fixing the problem that it can't be found with AssemblyDirectory at all because single-file has it in ApplicationDirectory?

Reply:
Isn't https://github.com/dotnet/runtime/issues/107040 about making it look in rather _fewer_ locations and this is fixing the problem that it can't be found with AssemblyDirectory at all because single-file has it in ApplicationDirectory?

Reply:
Isn't https://github.com/dotnet/runtime/issues/107040 about making it look in rather _fewer_ locations and this PR is fixing the problem that it can't be found with AssemblyDirectory at all (because in single-file case, it resides in ApplicationDirectory instead)?